src/data/energy.ts

```diff
@@ -612,26 +612,16 @@ export const getEnergyGasUnitCategory = (
 };
 
 export const getEnergyGasUnit = (
-  hass: HomeAssistant,
   prefs: EnergyPreferences,
   statisticsMetaData: Record<string, StatisticsMetaData> = {}
 ): string | undefined => {
   for (const source of prefs.energy_sources) {
     if (source.type !== "gas") {
       continue;
     }
-    const entity = hass.states[source.stat_energy_from];
-    if (entity?.attributes.unit_of_measurement) {
-      // Wh is normalized to kWh by stats generation
-      return entity.attributes.unit_of_measurement === "Wh"
-        ? "kWh"
-        : entity.attributes.unit_of_measurement;
-    }
```

Comment:
This change should be done also in function `getEnergyGasUnitCategory` just above, but it's not clear to me how to get hold of the statistics metadata which is needed.

Reply:
Fixed in [375d870](https://github.com/home-assistant/frontend/pull/13454/commits/375d870c58b09801a7ed9b6464be8f14175b3af5)